kagami/webui/src/api/admin.ts

```diff
@@ -0,0 +1,170 @@
+import api from './api';
+
+export interface AdminLoginRequest {
+  username: string;
+  password: string;
+}
+
+export interface AdminLoginResponse {
+  token: string;
+}
+
+export interface WorkerResource {
+  cpu: number;
+  memory: number;
+  storage: number;
+}
+
+export interface Worker {
+  address: string;
+  status: 'pending' | 'active' | 'offline';
+  resources: WorkerResource;
+}
+
+export interface WorkerListResponse {
+  code: number;
+  data: {
+    workers: Worker[];
+  };
+  message: string;
+}
+
+// Accept Worker Response
+export interface AcceptWorkerResponse {
+  code: number;
+  data: {
+    worker: Worker;
+  };
+  message: string;
+}
+
+// Delete Worker Response
+export interface DeleteWorkerResponse {
+  code: number;
+  message: string;
+}
+
+// Provider Info
+export interface ProviderInfo {
+  name: string;        // provider 名称
+  method: string;      // provider 方法 (rsync等)
+  status: string;      // provider 状态
+  upstream_url: string; // 上游地址
+}
+
+// Resource Info
+export interface ResourceInfo {
+  name: string;           // 资源名称
+  status: string;         // 资源状态
+  providers: ProviderInfo[];
+}
+
+// Get Worker Resource Response
+export interface GetWorkerResourceResponse {
+  code: number;
+  message: string;
+  data: {
+    resources: ResourceInfo[];
+  };
+}
+
+// Add Provider Request
+export interface AddProviderRequest {
+  resource_name: string;
+  provider: {
+    name: string;
+    method: string;
+    upstream_url: string;
+  }
+}
+
+// Add Provider Response
+export interface AddProviderResponse {
+  code: number;
+  message: string;
+  data: {
+    provider: ProviderInfo;
+  }
+}
+
+export const adminLogin = (data: AdminLoginRequest) => {
+  return api.post<AdminLoginResponse>('/admin/login', data);
+};
+
+export const getWorkerList = async (): Promise<Worker[]> => {
+  try {
+    const response = await api.get<WorkerListResponse>('/admin/worker/list');
+    if (response.data.code === 46) {
+      return response.data.data.workers;
+    }
+    throw new Error(response.data.message);
+  } catch (error) {
+    console.error('Failed to fetch workers:', error);
```

Comment:
Consider move error handling logic to a common part (api.ts)

Reply:
Sure, I'll add it to my to-do.